src/bucket/BucketSnapshotManager.cpp

```diff
@@ -131,16 +131,10 @@ BucketSnapshotManager::maybeUpdateSnapshotInternal(
     // snapshot, so use a shared lock.
     std::shared_lock<std::shared_mutex> lock(mSnapshotMutex);
 
-    // First update current snapshot
-    if (!snapshot ||
-        snapshot->getLedgerSeq() != managerSnapshot->getLedgerSeq())
-    {
-        // Should only update with a newer snapshot
-        releaseAssert(!snapshot || snapshot->getLedgerSeq() <
-                                       managerSnapshot->getLedgerSeq());
-        snapshot = std::make_unique<BucketListSnapshot<BucketT> const>(
-            *managerSnapshot);
-    }
+    // Should only update on snapshot creation
+    releaseAssert(!snapshot);
```

Comment:
Given this assert, we always update the snapshot. This function would make more sense as a constructor of `SearchableBucketListSnapshotBase`